.github/workflows/gradle-build.yaml

```diff
@@ -0,0 +1,27 @@
+name: Gradle
+
+on: [push, pull_request]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up JDK 1.8
+        uses: actions/setup-java@v1
+        with:
+          java-version: 1.8
+      - name: Cache Gradle packages
+        uses: actions/cache@v2
+        with:
+          path: ~/.gradle/caches
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle') }}
+          restore-keys: ${{ runner.os }}-gradle
+      - name: Build with Gradle
+        run: ./gradlew --continue verifyGoogleJavaFormat build connectedCheck
+      - name: Publish snapshot
```

Comment:
This should be conditioned on push?

Reply:
The script itself checks to ensure that it only runs on pushes, so it would be redundant to include in the GitHub config as well. (And my preference in general is to put as little logic in the CI config as possible, since it ties us less tightly to using that particular CI provider).

Reply:
I think it's clearer to add this to config. That way your `on` and your conditions are all in one place. Separation of concerns..

Reply:
OK - no real harm in including it for some safety in depth. Keeping it in the script as well though.